### PR TITLE
Pr/entity browser/single select

### PR DESCRIPTION
**Pull request by ottrob**

### CODE REVIEW

File: packages/entity-browser/src/components/DetailView/DetailForm.js [hunk 1]
```diff
@@ -28,9 +28,24 @@ export const DetailForm = props => {
           )
         }
       } else {
-        if (props.entity.paths[field.name].value !== null) {
-          const type = props.entity.paths[field.name].value.type
-          result.push(<Field name={field.name} type={type} key={i} label={field.label} component={LabeledField}/>)
+        const fieldProperties = props.entity.paths[field.name]
+        if (fieldProperties && fieldProperties.value !== null) {
```
Comment (rzueger) [inline packages/entity-browser/src/components/DetailView/DetailForm.js]: what if `fieldProperties.value` is `undefined`? Maybe `!= null` should be used?

File: packages/entity-browser/src/components/DetailView/DetailForm.js [hunk 2]
```diff
@@ -28,9 +28,24 @@ export const DetailForm = props => {
...
+          if (fieldProperties.type === 'field') {
+            const type = props.entity.paths[field.name].value.type
+            result.push(<Field name={field.name} type={type} key={i} label={field.label} component={LabeledField}/>)
+          } else if (fieldProperties.type === 'entity') {
+            const v = fieldProperties.value.key
+            result.push(<Field name={field.name} type="single-select" key={i} label={field.label}
+              component={LabeledField} options={{store: [{value: v, label: fieldProperties.value.display}]}}/>)
+            console.log(fieldProperties.value)
```
Comment (rzueger) [inline packages/entity-browser/src/components/DetailView/DetailForm.js]: console.log() should be removed

File: packages/entity-browser/src/components/DetailView/LabeledField.js [hunk 3]
```diff
@@ -53,7 +54,8 @@ LabeledField.propTypes = {
   meta: React.PropTypes.shape({
     touched: React.PropTypes.bool,
     error: React.PropTypes.objectOf(React.PropTypes.string)
-  })
+  }),
+  options: React.PropTypes.object
```
Comment (rzueger) [inline packages/entity-browser/src/components/DetailView/LabeledField.js]: can we define the `options` prop type a bit more precisely (i.e. by using shape()).

File: packages/entity-browser/src/dev/rest-responses/gender.json [hunk 4]
```diff
@@ -0,0 +1,173 @@
+{
```
Comment (rzueger) [inline packages/entity-browser/src/dev/rest-responses/gender.json]: not formatted properly (space after `:`)

File: packages/entity-browser/src/modules/searchForm/sagas.js [hunk 5]
```diff
@@ -27,6 +27,7 @@ export function* initialize({payload}) {
 
   const relationEntities = yield formDefinition.filter(searchField =>
     searchField.type === 'ch.tocco.nice2.model.form.components.simple.MultiSelectBox'
+    || searchField.type === 'ch.tocco.nice2.model.form.components.simple.SingleSelectBox'
```
Comment (rzueger) [inline packages/entity-browser/src/modules/searchForm/sagas.js]: I would prefer:

```
[
  'ch...',
  'ch...'
].includes(searchField.type)

Comment (MethodenMann) [inline packages/entity-browser/src/modules/searchForm/sagas.js]: IE and some other browser dont support includes:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Check if babel transpiles "includes" or use IndexOf

File: packages/tocco-ui/src/EditableValue/example.js [hunk 6]
```diff
@@ -14,17 +18,34 @@ export default () => {
               <EditableValue
                 type="string"
                 value="Test String"
-                onChange={console.log}
+                onChange={log}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/example.js]: no need for function "log"
onChange={console.log} should do the job

File: packages/tocco-ui/src/EditableValue/typeEditors/SingleSelect.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+      options={props.options.store}
+    />
+  )
+}
+
+SingleSelect.propTypes = {
+  onChange: React.PropTypes.func,
+  value: React.PropTypes.string,
+  options: React.PropTypes.shape({
+    store: React.PropTypes.arrayOf(
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/SingleSelect.js]: In MultiSelect this is called "possibleValues" and here "store" please rename one or the other

File: packages/entity-browser/src/modules/searchForm/sagas.js (hunk 5 above)
Comment (ottrob) [inline packages/entity-browser/src/modules/searchForm/sagas.js]: The babel polyfill already supports Array.prototype.includes
https://babeljs.io/docs/usage/polyfill/

> ... This means you can use new built-ins like Promise or WeakMap, static methods like Array.from or Object.assign, instance methods like Array.prototype.includes, ...

File: packages/entity-browser/src/components/DetailView/DetailForm.js [hunk 8]
```diff
@@ -28,9 +28,23 @@ export const DetailForm = props => {
...
+        if (fieldProperties && fieldProperties.value != null) {
+          if (fieldProperties.type === 'field') {
+            const type = props.entity.paths[field.name].value.type
+            result.push(<Field name={field.name} type={type} key={i} label={field.label} component={LabeledField}/>)
+          } else if (fieldProperties.type === 'entity') {
+            const value = fieldProperties.value.key
+            result.push(<Field name={field.name} type="single-select" key={i} label={field.label}
+              component={LabeledField} options={{store: [{value, label: fieldProperties.value.display}]}}/>)
+          } else if (fieldProperties.type === 'entity-list') {
+            const values = fieldProperties.value
```
Comment (MethodenMann) [inline packages/entity-browser/src/components/DetailView/DetailForm.js]: const store = fieldProperties.value.map(v => ({value: v.key, label: v.display})

File: packages/entity-browser/src/components/DetailView/DetailForm.js [hunk 9]
```diff
@@ -28,9 +28,23 @@ export const DetailForm = props => {
           )
         }
       } else {
-        if (props.entity.paths[field.name].value !== null) {
-          const type = props.entity.paths[field.name].value.type
-          result.push(<Field name={field.name} type={type} key={i} label={field.label} component={LabeledField}/>)
+        const fieldProperties = props.entity.paths[field.name]
+        if (fieldProperties && fieldProperties.value != null) {
```
Comment (MethodenMann) [inline packages/entity-browser/src/components/DetailView/DetailForm.js]: refactor suggestion:

put this:

` result.push(<Field name={field.name} key={i} label={field.label}   component={LabeledField} />)`

outside of if/elseif and just determine the different props inside if and add with ...props/>

**Pull request merged**